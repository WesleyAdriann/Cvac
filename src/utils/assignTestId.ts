import { Platform } from 'react-native'

interface ITestID {
  accessible?: boolean
  testID: string
  accessibilityLabel?: string
}

type TComponentType =
  | 'View'
  | 'ScrollView'
  | 'TextInput'
  | 'Text'
  | 'SafeAreaView'
  | 'Pressable'
  | 'Modal'

export const assignTestId = (
  componentType: TComponentType,
  componentId: string
): ITestID => {
  const id = `${componentType}_${componentId}`
  return Platform.OS === 'android'
    ? {
        testID: id,
        accessible: true,
        accessibilityLabel: id
      }
    : {
        testID: id,
        accessible: false
      }
}
export const mockTestID = (
  componentType: TComponentType,
  componentId: string
): ITestID => assignTestId(componentType, componentId)
