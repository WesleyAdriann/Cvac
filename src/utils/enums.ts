export enum ECalendarsName {
  'kid' = 'Crian√ßa',
  'elder' = 'Idoso',
  'adult' = 'Adulto',
  'teen' = 'Adolescente'
}
