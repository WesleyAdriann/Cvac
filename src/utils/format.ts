export const formatDate = (value: string) => {
  const clearDate = value.replace(/\D/g, '')

  if (clearDate.length === 8) return value

  const day = clearDate.slice(0, 2)
  const month = clearDate.slice(2, 4)
  const year = clearDate.slice(4, 8)

  if (!month) return day
  if (!year) return `${day}/${month}`
  return `${day}/${month}/${year}`
}
