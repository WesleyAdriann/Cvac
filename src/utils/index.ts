export * from './assignTestId'
export * from './withProviders'
export * from './format'
export * from './logs'
