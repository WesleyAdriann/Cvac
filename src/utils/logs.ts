export const log = (TAG: string, ...args: any[]) => {
  if (__DEV__) console.log(`[${TAG}] `, ...args)
}
