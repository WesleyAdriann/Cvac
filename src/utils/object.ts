export const isEmpty = (obj: object | null) => {
  if (!obj) return true
  // eslint-disable-next-line no-unreachable-loop
  for (const key in obj) return false
  return true
}
