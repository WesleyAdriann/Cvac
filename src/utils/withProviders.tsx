import React from 'react'
import { Provider as PaperProvider } from 'react-native-paper'
import { ThemeProvider } from 'styled-components'

import { theme, paperTheme } from '../tokens'

export const TestsProviders: React.FC = ({ children }) => (
  <PaperProvider theme={paperTheme} >
    <ThemeProvider theme={theme}>
      {children}
    </ThemeProvider>
  </PaperProvider>
)
