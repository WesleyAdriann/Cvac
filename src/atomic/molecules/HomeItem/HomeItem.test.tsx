import React from 'react'
import { render, cleanup } from '@testing-library/react-native'

import { AllTheProviders as wrapper, mockTestID } from '~/utils'

import { HomeItem, IHomeItem } from './index'

describe('Molecules/HomeItem', () => {
  const props: IHomeItem = {
    testID: 'HomeItem',
    icon: 'calendar',
    onPress: jest.fn(),
    text: 'Calendario'
  }

  const elContainer = mockTestID('Pressable', props.testID!).testID

  const renderComponent = () => render(<HomeItem {...props} />, { wrapper })
  let component = renderComponent()

  beforeEach(() => {
    jest.clearAllMocks()
    cleanup()
    component = renderComponent()
  })

  it(`should render component #${elContainer}`, async () => {
    const sut = await component.findByTestId(elContainer)
    expect(sut).toBeTruthy()
  })

  it('should render snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot()
  })
})
