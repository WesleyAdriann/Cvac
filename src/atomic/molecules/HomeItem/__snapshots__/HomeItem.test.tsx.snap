// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules/HomeItem should render snapshot 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessible={false}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderColor": "#00A3C9",
          "borderRadius": 2,
          "borderStyle": "solid",
          "borderWidth": 1,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "height": 351,
          "justifyContent": "center",
          "marginRight": 0,
          "paddingBottom": 14,
          "paddingLeft": 14,
          "paddingRight": 14,
          "paddingTop": 14,
        },
      ]
    }
    testID="Pressable_HomeItem"
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#00C9D2",
            "fontSize": 12,
          },
          Array [
            Object {
              "fontSize": 28,
            },
          ],
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      ó°ƒ­
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#424242",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Array [
            Object {
              "color": "#00C9D2",
              "fontSize": 16,
              "textAlign": "center",
              "textTransform": "uppercase",
            },
          ],
        ]
      }
    >
      Calendario
    </Text>
  </View>
</View>
`;
